### DISCUSSION TITLE
About Routing

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#58] [Off topic]
May I ask what Routing architecture/package do you use?

Do you use your own declarative implementation as routing?

Is it correct to change route (imperative way) inside a `RearchConsumer` `build()` method?

**Comment by GregoryConrad** [comment]
I personally have used auto_route because it’s less boilerplate and is loaded with features but go_router also seems like a widely accepted solution.

it would be cool to have a rearch_router, but I don’t have the time nor motivation to learn the mess that is the Flutter Navigation 2.0 API to make a dedicated package for it.

it is perfectly acceptable to change routes in a callback in the build method. I’d suggest putting your router in its own capsule, and then actions/functions to navigate to particular routes in other capsule(s) that use your router capsule, but you can also manually handle navigation in build method callbacks for smaller projects.

**Reply by busslina** [reply]
Thanks ;)
